lib/foreman_maintain/dependency_graph.rb

```diff
@@ -0,0 +1,76 @@
+require 'tsort'
+
+module ForemanMaintain
+  class DependencyGraph
+    include TSort
+
+    attr_reader :graph, :collection
+
+    def self.sort(collection)
+      new(collection).tsort.map(&:ensure_instance)
```

Comment:
If I read this properly, it would mean we would loose the original objects and re-create them again. This goes against
the intention of being able to add parametrized steps in the scenarios. Think of this example:

```ruby
# in scenario
add_step(Procedures::Packages::Install.new(:packages => 'hdparm'))
```

After the sorting, we would loose the `:packages` parameter